tests/test_datastructures.py

```diff
@@ -128,3 +128,50 @@ def test_set_fail(self):
         t = self.TestDict()
         with self.assertRaises(exceptions.FieldError):
             t['stringval'] = self.FAILVAL
+
+
+class ValidatingDependentDictItemsTest(unittest.TestCase):
+    GOOD_INT_VALUE = 1
+    GOOD_STR_VALUE = '1'
+    WRONG_VALUE = b'wrong_value'
+
+    class TestDependentItemDict(datastructures.ValidatingDependentDictItems):
+        ITEMS = {
+            'intvalue': 0,
+            'stringvalue': ''
+        }
+
+        DEPENDENT_ITEMS = [
+            ('intvalue', 'stringvalue'),
+        ]
+
+        def validate_intvalue_stringvalue(self, value):
```

Comment:
Quite frankly, this validation tests two fields separately. Maybe you could make some validation, which would test interaction between these two fields?
I.e.:
```python
def validate_intvalue_stringvalue(self, value):
    if value['intvalue'] != int(value['stringvalue']):
        raise exceptions.FieldError()
```